src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/ImplementStandardExceptionConstructors.Fixer.cs

```diff
@@ -60,7 +60,10 @@ private static async Task<Document> AddConstructorsAsync(Document document, IEnu
             CodeAnalysis.Text.TextSpan diagnosticSpan = diagnostics.First().Location.SourceSpan; // All the diagnostics are reported at the same location -- the name of the declared class -- so it doesn't matter which one we pick
             SyntaxNode node = root.FindNode(diagnosticSpan);
             SyntaxNode targetNode = editor.Generator.GetDeclaration(node, DeclarationKind.Class);
-            var typeSymbol = model.GetDeclaredSymbol(targetNode) as INamedTypeSymbol;
+            if (!(model.GetDeclaredSymbol(targetNode) is INamedTypeSymbol typeSymbol))
+            {
+                return document;
```

Comment:
💭 Maybe throw since this is such an unexpected case?

```suggestion
                throw new InvalidOperationException();
```

Reply:
I prefer to keep the code fixes as conservative as possible :)